definitions/checks/system_registration.rb

```diff
@@ -5,15 +5,26 @@ class Checks::SystemRegistration < ForemanMaintain::Check
     tags :default
 
     confine do
-      file_exists?('/etc/rhsm/rhsm.conf') &&
-        !feature(:foreman_server) &&
-        feature(:foreman_proxy)
+      file_exists?('/etc/rhsm/rhsm.conf')
     end
   end
 
   def run
     if rhsm_hostname_eql_hostname?
-      warn! 'System is self registered'
+      if feature(:downstream)
+        notification_message = <<-MESSAGE
+  Satellite is self registered
+
+  Self registered satellite won't be supported as of Satellite version 6.3. Please follow following documentation to resolve.
+
+  https://access.redhat.com/documentation/en-us/red_hat_satellite/6.3/html/installation_guide/preparing_your_environment_for_installation#system_requirements
+
+  https://access.redhat.com/documentation/en-us/red_hat_satellite/6.3/html/upgrading_and_updating_red_hat_satellite/upgrading_red_hat_satellite#migrating_a_self_registered_satellite
+    MESSAGE
```

Comment:
As per old PR it has been suggested to move fail! method outside of if else block so we always fail if Satellite is self registered.